tests/jquery.flot.composeImages.Test.js

```diff
@@ -0,0 +1,698 @@
+        '#canvasSource1 {position:relative; left:-180px; top:-10px;}' +
+        '</style>' +
+        '<div id="test-container" style="width: 600px;height: 400px">' +
+        '<svg class="imgsrc" id="svgSource1" viewBox="0 0 250 150" xmlns="http://www.w3.org/2000/svg" width="250" height="150" title="svg1">' +
+            '<circle class="externalCSS" id="c1" cx="230" cy="20" r="15" style="fill:red"/>' +
+            '<circle class="externalCSS" id="c2" cx="175" cy="100" r="25" style="fill:#00FF00"/>' +
+            '<circle class="externalCSS" id="c3" cx="50" cy="130" r="40" style="fill:blue"/>' +
+        '</svg>' +
+        '<canvas class="imgsrc" id="canvasSource1" width="20" height="20" title="canvasSource1"></canvas>' +
+        '</div>' +
+        '<canvas id="myCanvas" width="150" height="150" style="border:1px solid #d3d3d3;"></canvas>'
+        ).find('.imgsrc').toArray();
+
+        var originalCanvas1 = document.getElementById("canvasSource1");
+        var destinationCanvas = document.getElementById("myCanvas");
+        var canvas1_Data; //used later
+        var canvas3_Data; //used later
+        var pixelData;//used later
+
+        drawARectangleOnCanvas(originalCanvas1, "#FF0000");
+
+        composeImages(sources, destinationCanvas).then(function() {
+            expect(destinationCanvas.width).toBe(250);
+            expect(destinationCanvas.height).toBe(150);
+
+            canvas1_Data = originalCanvas1.getContext('2d').getImageData(0, 0, 20, 20).data;
+            canvas3_Data = destinationCanvas.getContext('2d').getImageData(250 - 180 + 0, 150 - 10 - 20, 20, 20).data; //250 - 180 + 4
+            expect(matchPixelDataArrays(canvas1_Data, canvas3_Data)).toBe(true);
+
+
+            //circle centers
+            pixelData = destinationCanvas.getContext('2d').getImageData(230, 20, 1, 1).data;
+            expect(matchPixelColor(pixelData, 255, 0, 0, 255)).toBe(true);
+
+            pixelData = destinationCanvas.getContext('2d').getImageData(175, 100, 1, 1).data;
+            expect(matchPixelColor(pixelData, 0, 255, 0, 255)).toBe(true);
+
+            pixelData = destinationCanvas.getContext('2d').getImageData(50, 130, 1, 1).data;
+            expect(matchPixelColor(pixelData, 0, 0, 255, 255)).toBe(true);
+
+            //other points on circles should match the required colors, because of configured diameters
+            pixelData = destinationCanvas.getContext('2d').getImageData(220, 17, 1, 1).data;
+            expect(matchPixelColor(pixelData, 255, 0, 0, 255)).toBe(true);
+
+            pixelData = destinationCanvas.getContext('2d').getImageData(190, 114, 1, 1).data;
+            expect(matchPixelColor(pixelData, 0, 255, 0, 255)).toBe(true);
+
+            pixelData = destinationCanvas.getContext('2d').getImageData(80, 149, 1, 1).data;
+            expect(matchPixelColor(pixelData, 0, 0, 255, 255)).toBe(true);
+            expect(pixelData).toMatchPixelColor([0, 0, 255, 255]);
```

Comment:
Nice! I think you can push this even further and use one more helper method something like:
```js
expect(pixelColor(canvas, 80, 149, 1, 1)).toMatchPixelColor([0, 0, 255, 255]);
```

Reply:
done